src/file.ts

```diff
@@ -1752,12 +1754,20 @@ class File extends ServiceObject<File> {
       md5,
     });
 
+    const progressStream = new ProgressStream();
+    if (options.onUploadProgress) {
+      progressStream.on('progress', bytesRead => {
+        options.onUploadProgress!({bytesRead});
+      });
+    }
+
     const fileWriteStream = duplexify();
 
     const stream = streamEvents(
       pumpify([
         gzip ? zlib.createGzip() : through(),
         validateStream,
+        progressStream,
```

Comment:
Referring back to an old comment (https://github.com/AVaksman/nodejs-storage/pull/12#discussion_r364463846), was it possible to move this below the write stream?

Reply:
With new changes `prgressStream` is not used any more.